src/features/filter.ts

```diff
@@ -1,10 +1,47 @@
-export const actions = { /* put action creators here */};
+import { Status } from '../types/Status';
 
-const filterReducer = () => {
-  return {
-    query: '',
-    status: 'all',
-  };
+type SetQuery = {
+  type: 'query/SET',
+  payload: string,
+};
+
+type SetStatus = {
+  type: 'status/SET',
+  payload: Status,
+};
+
+const defaultState = {
+  query: '',
+  status: 'All' as Status,
+};
+
+const setStatus = (status: Status): SetStatus => ({
+  type: 'status/SET',
+  payload: status,
+});
+
+const setQuery = (query: string): SetQuery => ({
+  type: 'query/SET',
+  payload: query,
+});
+
+export const actions = { setQuery, setStatus };
```

Comment:
Consider grouping exported objects into a single export statement at the end of the file for better organization and to follow best practices.